applications/verifiable_offchain_computing.md

```diff
@@ -0,0 +1,241 @@
+
+Consistent state management is crucial when developing an application that communicates with a blockchain. Developers need a reliable and transactional way to perform operations, where read and write tasks are combined into a single unit and executed atomically on the blockchain. This approach aligns with the ACID principle found in transactional database management systems.
+
+The sdk is here to simplify development by providing a stable, ACID-compliant connection with ink! smart contracts. This eliminates concurrency issues and enables a request-response programming model for seamless interaction between the workers and on-chain smart contracts.
+
+##### Key Benefits
+- Seamless connectivity between ink! smart contracts and workers
+- Reliable KV-store on blockchains for durable state management
+- Transactional (ACID) on-chain KV-store designed for stateful dApps deployed on Phala Cloud
+- ACID-compliant read, write, and contract call operations for consistent data handling
+- Request-response programming model that simplifies interactions between worker and on-chain smart contracts
+
+##### Architectural Diagram
+
+Ink! smart contracts, built with ink! v6, will be deployed on `pallet-revive` and an application, built on js/ts, will be deployed on Phala Cloud.
+
+Websocket and http requests will be used to communicate between the ink! smart contracts and the application on Phala Cloud.
+
+Our sdk will manage the synchronization mechanism and the developers would have a reliable and transactional way to perform operations, where read and write tasks are combined into a single unit and executed atomically on the blockchain.
+
+The SDK will be divided into two parts:
+- the client will be integrated as a Rust library within ink! smart contracts.
+- the rollup will be integrated as a JavaScript library within applications deployed on Phala Cloud.
+
+![communication-contracts.png](../assets/verifiable_offchain_computing/communication-contracts.png)
+
+##### Note
+In 2023, our team built the [Phala's offchain rollup anchor](https://github.com/Phala-Network/phat-offchain-rollup) implementation for ink! smart contracts in the context of Phala Builder.
+Even if this sdk is still used in mainnet, it was built with ink! v4, OpenBrush and Phat Contracts. OpenBrush is now deprecated. Phat Contracts are replaced by Phala Cloud. 
+The main goal is to provide the same features as [Phala's offchain rollup anchor](https://github.com/Phala-Network/phat-offchain-rollup) but with the latest technologies (ink! v6 and Phala Cloud compatibility)
+
+##### Languages
+The library/crate used in ink! smart contracts will be written in Rust and the latest released ink! version.
+The library used in the worker deployed on Phala Cloud will be written in TypeScript and will use the latest version of Polkadot.js api.
```

Comment:
@GuiGou12358 It looks like Polkadot.js API is essentially in maintenance mode ([see this for example](https://forum.polkadot.network/t/polkadot-js-moving-forward/6235/9)), can you use something like [Polkadot-API](https://docs.polkadot.com/develop/toolkit/api-libraries/papi/) instead ([see also](https://papi.how/))?

Reply:
I was not aware that Polkadot.js API was switched in maintenance mode. Good to know! So I will use Polkadot-API.